### DISCUSSION TITLE
How to provide an 'initialTab' to AutoTabsRouter

### DISCUSSION THREAD

**Discussion by Adam-Langley** [Milad-Akarie/auto_route_library#1711] [Q&A] [answered]
I understand there may already be a way to do this - using 'initial' route - but it doesn't quite seem to fit the bill.

I have a master/detail view, on the left is a list, on the right is an AutoTabsRouter.
For example, imagine a customers list on the left, and 2 customer tabs on the right, 'Profile', and 'Order History'.
When I select a customer in the list, the initial route sets the tab index to 0 on the right... 'Profile'.
This is correct - however, I would also like the state of the tabs to be retained when moving _between_ 'Customers' - i.e. when the user tabs to 'Order History', and then selects the next customer in the list, then the active tab should remain on 'Order History' for the newly selected customer.

I don't think 'initial' in the route declaration is the right thing, because these are cached - they dont get re-evaluated.

It simply feels like 'AutoTabsRouter' should have an initialTab route... but I also feel like this may conflict with the 'initial' during deep linking...

What's the right way to achieve this?

Thanks!

**Comment by Milad-Akarie** [comment, marked as answer]
Hey Adam, nice hearing from you again.
Because of deep-linking functionality AutoTabsRouter does not have an initial index property, the initial flag in declaration is more like a default configuration (if no child pass), so if you want to your tab-hosting route to start with a certain tab just pass it as a child route e.g

```dart
router.push(TabHostRoute(children:[TargetTabRoute()]));
```

**Reply by Adam-Langley** [reply]
Perfect!
Darn, I remember seeing this a while back - looks like I failed to retain this critical piece of info. Seems obvious now however.
Thank you!
